### PR TITLE
avoid literal quoting in config lists

### PR DESCRIPTION
**Pull request by bcoca**
##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

config

### CODE REVIEW

File: lib/ansible/config/manager.py [hunk 1]
```diff
@@ -92,7 +92,16 @@ def ensure_type(value, value_type, origin=None):
 
         elif value_type == 'list':
             if isinstance(value, string_types):
-                value = [x.strip() for x in value.split(',')]
+                list_value = []
+                for item in value.split(','):
+                    item = item.strip()
+                    # in case this came as a 'string of quoted items' (ini, env var)
+                    for q in ('"', "'"):
```
Comment (s-hertel) [inline lib/ansible/config/manager.py]: You could use `unquote` like string options to just remove the outer quotation marks. This removes all instances of it.

Comment (bcoca) [reply lib/ansible/config/manager.py]: only if i were smart ...

Comment (s-hertel) [reply lib/ansible/config/manager.py]: was one of those days for me, I'm blaming it on monday :stuck_out_tongue_closed_eyes:

File: lib/ansible/plugins/__init__.py [hunk 2]
```diff
@@ -66,6 +66,7 @@ def get_options(self, hostvars=None):
         defs = C.config.get_configuration_definitions(plugin_type=get_plugin_class(self), name=self._load_name)
         for option in defs:
             options[option] = self.get_option(option, hostvars=hostvars)
+        print(options)
```
Comment (s-hertel) [inline lib/ansible/plugins/__init__.py]:
```suggestion

```

File: test/integration/targets/config/lookup_plugins/types.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+from ansible.plugins.lookup import LookupBase
+
+
+class LookupModule(LookupBase):
+
+    def run(self, terms, variables=None, **kwargs):
+
+        self.set_options(var_options=variables, direct=kwargs)
+        print(list(enumerate(self.get_options())))
```
Comment (samdoran) [inline test/integration/targets/config/lookup_plugins/types.py]:
```suggestion
```

**Review by samdoran — CHANGES_REQUESTED** [review]
The fix looks good. The new playbook with the test config needs to be added to `test/integration/targets/config/runme.sh` so that it will run.

```patch
diff --git test/integration/targets/config/runme.sh test/integration/targets/config/runme.sh
index bbff6acc3e..d19c3fa8ed 100755
--- test/integration/targets/config/runme.sh
+++ test/integration/targets/config/runme.sh
@@ -19,5 +19,7 @@ ANSIBLE_CONFIG=nonexistent.cfg ansible-config dump --only-changed -v | grep 'No
 # https://github.com/ansible/ansible/pull/73715
 ANSIBLE_CONFIG=inline_comment_ansible.cfg ansible-config dump --only-changed | grep "'ansibull'"
 
+ANSIBLE_CONFIG=type_munging.cfg ansible-playbook types.yml "$@"
+
 # test the config option validation
 ansible-playbook validation.yml "$@"
```

**Review by samdoran — APPROVED** [review]

**Pull request merged**